src/WalletRpcPlugin.ts

```diff
@@ -0,0 +1,152 @@
+
+  public constructor() {
+    super();
+  }
+
+  /**
+   * Request to add a new chain to the user's wallet.
+   *
+   * See [EIP-3085](https://eips.ethereum.org/EIPS/eip-3085) for more details.
+   *
+   * @param param - See {@link AddEthereumChainRequest}
+   * @returns a Promise that resolves if the request is successful
+   */
+  public async addEthereumChain(param: AddEthereumChainRequest): Promise<void> {
+    return this.requestManager.send({
+      method: "wallet_addEthereumChain",
+      params: [
+        {
+          ...param,
+          chainId: utils.toHex(param.chainId),
+        },
+      ],
+    });
+  }
+
+  /**
+   * Switch to a new chain and register it with the user’s wallet if it isn’t already recognized.
+   *
+   * See [EIP-2015](https://eips.ethereum.org/EIPS/eip-2015) for more details.
+   *
+   * @param param - See {@link UpdateEthereumChainRequest}
+   * @returns a Promise that resolves if the request is successful
+   */
+  public async updateEthereumChain(
+    param: UpdateEthereumChainRequest
+  ): Promise<void> {
+    return this.requestManager.send({
+      method: "wallet_updateEthereumChain",
+      params: [
+        {
+          ...param,
+          chainId: utils.toHex(param.chainId),
+        },
+      ],
+    });
+  }
+
+  /**
+   * Switch the wallet’s currently active chain.
+   *
+   * See [EIP-3326](https://eips.ethereum.org/EIPS/eip-3326) for more details.
+   *
+   * @param param - See {@link SwitchEthereumChainRequest}
+   * @returns a Promise that resolves if the request is successful
+   */
+  public async switchEthereumChain(
+    param: SwitchEthereumChainRequest
+  ): Promise<void> {
+    return this.requestManager.send({
+      method: "wallet_switchEthereumChain",
+      params: [
+        {
+          ...param,
+          chainId: utils.toHex(param.chainId),
+        },
+      ],
+    });
+  }
+
+  /**
+   * Return a list of owned assets for the given address.
+   *
+   * See [EIP-2256](https://eips.ethereum.org/EIPS/eip-2256) for more details.
+   *
+   * @param param - See {@link GetOwnedAssetsRequest}
+   * @returns a Promise that resolves to a list of owned assets, see {@link GetOwnedAssetsResult}
+   */
+  public async getOwnedAssets(
+    param: GetOwnedAssetsRequest
+  ): Promise<GetOwnedAssetsResult> {
+    validator.validator.validate(["address"], [param.address]);
+
+    const trueParam = { ...param };
+    if (trueParam.options?.chainId) {
+      trueParam.options.chainId = utils.toHex(trueParam.options.chainId);
+    }
+
+    const result = await this.requestManager.send({
+      method: "wallet_getOwnedAssets",
+      params: [trueParam],
+    });
+
+    return parseToGetOwnedAssetsResult(result);
+  }
+
+  /**
+   * Add an asset to the user's wallet.
+   *
+   * See [EIP-747](https://eips.ethereum.org/EIPS/eip-747) for more details.
+   *
+   * @param param - See {@link WatchAssetRequest}
+   * @returns a Promise that resolves if the request is successful
```

Comment:
Do you know why this `Promise` resolves to a `boolean` while the other resolve to `void`?

```suggestion
   * @returns a Promise that resolves to `true` if the request is successful
```